server/config/config.go

```diff
@@ -107,10 +109,25 @@ func (conf *Config) Reconfigure(stored StoredConfig, mmconf *model.Config, licen
 		return err
 	}
 
+	listenAddress := mmconf.ServiceSettings.ListenAddress
+	if listenAddress == nil {
+		return errors.New("plugin requires Mattermost Listen Address to be set")
+	}
+	host, port, err := net.SplitHostPort(*listenAddress)
+	if err != nil {
+		return err
+	}
+
+	if host == "" {
+		host = "127.0.0.1"
+	}
+	listenURL := "http://" + host + ":" + port
```

Comment:
Will we run into any certificate complications when the server is configured to forward `80 -> 443`? For example, the server's SSL cert may not match the `host` we are using here. Though I could be way wrong here.

Reply:
@mickmister What certificate? (`http`, not `https`)

@hanzei nit 3/5 s/listenURL/loopbackURL/, listen(Address-not-URL) should be ":8065", on the server.

Reply:
@mickmister Good question. The `Forward80To443` setting isn't an issue as it only adds an additional listener to port 80. The caveat is in `ConnectionSecurity`. If enabled, the MM server takes care of terminating the TLS connections. In that case there using the SiteURL is preferred as indeed the host name could mismatch.